src/lib.rs

```diff
@@ -53,3 +52,37 @@ impl Behaviour {
         }
     }
 }
+
+pub async fn load_secret_key_from_file(path: impl AsRef<Path> + Debug) -> Result<SecretKey> {
+    let bytes = fs::read(&path)
+        .await
+        .context(format!("No secret file at {:?}", path))?;
+    let secret_key = SecretKey::from_bytes(bytes)?;
+    Ok(secret_key)
+}
+
+pub async fn generate_secret_key_file(path: PathBuf) -> Result<SecretKey> {
+    if let Some(parent) = path.parent() {
+        DirBuilder::new()
+            .recursive(true)
+            .create(parent)
+            .await
+            .context(format!(
+                "Could not create directory for secret file: {:?}",
+                parent
+            ))?;
```

Comment:
[`tokio::fs::create_dir_all`](https://docs.rs/tokio/1.8.0/tokio/fs/fn.create_dir_all.html) is quite a bit shorter no? :)